monitor/plugins/spark_progress.py

```diff
@@ -30,58 +30,103 @@
+                application_progress_error['timestamp'] = time.time() * 1000
+                application_progress_error['dimensions'] = self.dimensions
+                
+                self.monasca.send_metrics([application_progress_error])
+                
+            time.sleep(MONITORING_INTERVAL)
+
 
     def _get_elapsed_time(self, gmt_timestamp):
         try:
             local_tz = tzlocal.get_localzone()
+
         except Exception as e:
             local_tz = "America/Recife"
             local_tz = pytz.timezone(local_tz)
+
         date_time = datetime.strptime(gmt_timestamp, '%Y-%m-%dT%H:%M:%S.%fGMT')
         date_time = date_time.replace(tzinfo=pytz.utc).astimezone(local_tz)
-        elapsed_time = datetime.now() - date_time.replace(tzinfo=None)
-        return elapsed_time.seconds
-
-    def _publish_measurement(self, job_request):
-
-        application_progress_error = {}
-
-        for result in job_request.json():
-
-            progress = result['numCompletedTasks'] / float(result['numTasks'])
-            progress = float("{:10.4f}".format(progress))
-            # Add to metric_info values for this measurement:
-            # value and timestamp
-            ref_value = self._get_elapsed_time(result['submissionTime']) / float(self.expected_time)
-            error = progress - ref_value
-            # The Monasca metric must have the 3 following fields to be created: name, value and timestamp,
-            # but also is possible to increment the metrics identities - using dimensions -  and informations -
-            # using value_meta, a dictionary that contains aditional information about the measurement  .
-            application_progress_error['name'] = 'application-progress.error'
-            application_progress_error['value'] = error
-            application_progress_error['timestamp'] = time.time() * 1000
-            application_progress_error['dimensions'] = self.dimensions
-            # Sending the metric to Monasca
-            self.monasca.send_metrics([application_progress_error])
+        date_time = date_time.replace(tzinfo=None)
+        datetime_now = datetime.now()
+        elapsed_time = datetime_now - date_time
 
+        return elapsed_time.seconds
 
-            print "Metric successfully published"
-
-            time.sleep(MONITORING_INTERVAL)
 
     def monitoring_application(self):
         try:
-            
-            job_request = requests.get(self.submission_url + ':4040/api/v1/applications/' + self.app_id + '/jobs')
-            
+
+            job_request = requests.get(self.submission_url
+                          + ':4040/api/v1/applications/'
+                          + self.app_id + '/jobs')
+
             self._publish_measurement(job_request)
 
         except Exception as ex:
-            print "Error: No application found for %s. %s remaining attempts" % (self.app_id, self.attempts)
+            print ("Error: No application found for %s. %s remaining attempts"
```

Comment:
self.attempts is not updated anywhere. I think you should remove it from this message.

Reply:
This is from the last version of monitor. The idea of this pull request is implement the spark progress with multiple jobs.

Reply:
Add a FIXME comment, please. So that we can remember to change this in the near future.